### ISSUE TITLE
qhp file corrupt when subsection with no parent section

### ISSUE THREAD

**Issue by ferdymercury** [doxygen/doxygen#8753]
**Describe the bug**
Consider this markdown page:

```
\mainpage %Test Page

### MySmallTitle
hello here

### AnotherTitle
hello there
```
Doxygen issues this warning,
```
mainpage.md:4: warning: found subsubsection command outside of subsection context!
mainpage.md:7: warning: found subsubsection command outside of subsection context!
```

but then, the QHP generation fails:
```
Error in line 14: Opening and ending tag mismatch.
error: failed to run qhelpgenerator on index.qhp
```

**Expected behavior**
Doxygen generates a valid QHP file even if there is a subsubsection with no parent section.

**To Reproduce**
Add the snippet above to a mainpage.md file, and add it as `INPUT` in a `Doxyfile`. Set `GENERATE_QHP` to `YES`

**Version**
1.9.2 built from sources  

**Additional context**
Partially related https://github.com/doxygen/doxygen/issues/5880
https://github.com/doxygen/doxygen/issues/5892
https://github.com/root-project/root/issues/8781#issuecomment-903547557

**Comment by albert-github** [comment]
Looks like there are 2 `</section>`s too many, this is independent on the number of sections beforehand as when I use:
```
\mainpage %Test Page

### MySmallTitle
hello here
```
there are 2 `</section>`s too many as well.

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #8754

**Comment by ferdymercury** [comment]
Thanks for the swift reply and fix!!

**Comment by albert-github** [comment]
Code has been integrated in master on GitHub (please don't close the issue as this will be done at the moment of an official release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
